### ISSUE TITLE
[PRELUDE] Compilation error. currently_rule variable not declared at w_writer_log_thread function

### ISSUE THREAD

**Issue by MrRubi** [wazuh/wazuh#2552]
|Wazuh version|Install type|Install method|Platform|
|---|---|---|---|
| 3.7.0 | Manager | Sources | Ubuntu 16.04.5 LTS |

Hello,

Trying to install a wazuh instance that will fork the events to a prelude instance I found a compilation error when I was manually installing wazuh. 

The error is described below:

Te function w_writer_log_thread placed at src/analysisd/analysisd.c has an ifdef for prelude 

```
    #ifdef PRELUDE_OUTPUT_ENABLED
                /* Log to prelude */
                if (Config.prelude) {
                    if (Config.prelude_log_level <= currently_rule->level) {
                        OS_PreludeLog(lf);
                    }
                }
#endif
```

The problem is that when you complie this file, it complains because the "currently_rule" variable has not been previosuly defined. As you can see in the complete funcion definition

```
void * w_writer_log_thread(__attribute__((unused)) void * args ){
    Eventinfo *lf;

    while(1){
            /* Receive message from queue */
            if (lf = queue_pop_ex(writer_queue_log), lf) {

                w_mutex_lock(&writer_threads_mutex);
                w_inc_alerts_written();

                if (Config.custom_alert_output) {
                    __crt_ftell = ftell(_aflog);
                    OS_CustomLog(lf, Config.custom_alert_output_format);
                } else if (Config.alerts_log) {
                    __crt_ftell = ftell(_aflog);
                    OS_Log(lf);
                } else if(Config.jsonout_output){
                    __crt_ftell = ftell(_jflog);
                }
                /* Log to json file */
                if (Config.jsonout_output) {
                    jsonout_output_event(lf);
                }

    #ifdef PRELUDE_OUTPUT_ENABLED
                /* Log to prelude */
                if (Config.prelude) {
                    if (Config.prelude_log_level <= currently_rule->level) {
                        OS_PreludeLog(lf);
                    }
                }
    #endif

    #ifdef ZEROMQ_OUTPUT_ENABLED
                /* Log to zeromq */
                if (Config.zeromq_output) {
                    zeromq_output_event(lf);
                }
    #endif
                w_mutex_unlock(&writer_threads_mutex);
                Free_Eventinfo(lf);
            }
    }
}
```
In my case a did a workaround to make it work

```
void * w_writer_log_thread(__attribute__((unused)) void * args ){
    Eventinfo *lf;
    ADDED --> RuleInfo *currently_rule = NULL;

    while(1){
            /* Receive message from queue */
            if (lf = queue_pop_ex(writer_queue_log), lf) {

                ADDED --> currently_rule = lf->generated_rule;
                w_mutex_lock(&writer_threads_mutex);
                w_inc_alerts_written();

                if (Config.custom_alert_output) {
                    __crt_ftell = ftell(_aflog);
                    OS_CustomLog(lf, Config.custom_alert_output_format);
                } else if (Config.alerts_log) {
                    __crt_ftell = ftell(_aflog);
                    OS_Log(lf);
                } else if(Config.jsonout_output){
                    __crt_ftell = ftell(_jflog);
                }
                /* Log to json file */
                if (Config.jsonout_output) {
                    jsonout_output_event(lf);
                }

    #ifdef PRELUDE_OUTPUT_ENABLED
                /* Log to prelude */
                if (Config.prelude) {
                    if (Config.prelude_log_level <= currently_rule->level) {
                        OS_PreludeLog(lf);
                    }
                }
    #endif

    #ifdef ZEROMQ_OUTPUT_ENABLED
                /* Log to zeromq */
                if (Config.zeromq_output) {
                    zeromq_output_event(lf);
                }
    #endif
                w_mutex_unlock(&writer_threads_mutex);
                Free_Eventinfo(lf);
            }
    }
}
```

I detected this error at v3.7.0 but is still present at v3.8.2

The installation was performed manually (make clean, make all, make TARGET=server) in an Ubuntu 16.0.4 LTS as a manager.

Thank you
Regards

**Comment by Zenidd** [comment]
Hi @MrRubi,

First of all, sorry for the late response. 
We have recreated the environment and as you explain to us, the compilation fails if the PRELUDE flag is set to 1.

**Output compiling with `# make TARGET=server USE_PRELUDE=1 -j4`** (this is the compile command we recommend, you should avoid using `make all`):

```
analysisd/analysisd.c: In function ‘w_writer_log_thread’:
analysisd/analysisd.c:1757:53: error: ‘currently_rule’ undeclared (first use in this function); did you mean ‘current_time’?
                     if (Config.prelude_log_level <= currently_rule->level) {
                                                     ^~~~~~~~~~~~~~
                                                     current_time
analysisd/analysisd.c:1757:53: note: each undeclared identifier is reported only once for each function it appears in
Makefile:1331: recipe for target 'analysisd/analysisd-test.o' failed
make[1]: *** [analysisd/analysisd-test.o] Error 1
make[1]: se sale del directorio '/home/vagrant/wazuh/wazuh/src'
Makefile:541: recipe for target 'server' failed

```
<br></br>
Well, thank you for your contribution, we will review this change and incorporate it as soon as possible in our project.
<br></br>

Best regards,

Juan Pablo Sáez

**Comment by vikman90** [comment]
@MrRubi this issue has finally been fixed, merging PR's #5003 y and #5004 by @fpoirotte into branches 3.14 and master.

Thank you all for this contribution!
Best regards.
